### ISSUE TITLE
Embed a web page into the Epic home page??

### ISSUE THREAD

**Issue by AnaHotaling** [realbig/RBM_TH_EPIC#61]
Eva really wants the public to know about how Epic gives back to the community, and I agree! We don't want to be blatant, like the Gazelle Girl site, which pretty much screams that it raised $90,000 for charity. We want people to see all the ways Epic has helped our community... and offer visitors a chance to help out by donating. One thought is to embed our Epic Races Crowdrise page into the space at the home page where the huge Crowdrise widget is right now.

Another thought is to take out that huge widget and custom tailor a grid of sorts, into which each event's Crowdrise widget can be plugged in. This would allow visitors to see all the Crowdrise widgets on the front page and click the one they want. 

I asked Eva if this was fine and she said yes. Then I got caught up in marathon stuff and just found out, more than a week later, that I never submitted this!!! Ugh.

**Comment by brashrebel** [comment]
I believe you're opening this up here now for discussion purposes, correct? If you're asking for something specific to be done it isn't clear yet. It sounds like you're trying to make a decision about how best to display some new information. Can I ask some questions?
1. Could you simply blog about Epic's giving and include prominent links to donate opportunities? If not enough people read the blog you could send it to your newsletter list and we could discuss ways to make the blog more prominent. You could also share the post on social media and it would show in the Epic News widget on the home page.
2. Instead of embedding something, could we just design some nice looking button or new widget that describes and links to this information? Embedding from an external source can be problematic for a number of reasons but one big reason is that it will not look like your site. It will usually look a bit out of place and unprofessional. That's because you're simply pulling information from a different website that will be styled like _that_ site and not _yours_. I'd hate to see the Epic Races home page get cluttered and lose its cohesive style.
3. If we are going to add something new to the home page, do we really need to _replace_ something? There aren't a limited number of spaces. We can keep going down the page almost infinitely. So something new could be placed above or below what's there now.
4. Oh, now I'm getting what you said about replacing the _widget_ with the _page_. Ummm...like I said in 2 above, embedding an external page will look bad and there's really not much you can do about it. It is a big step towards a Frankenstein page where some elements very obviously don't belong and are not cohesive with the rest. I'd recommend using caution here. If there is information you'd like to present to visitors beyond what is there, let's talk about it. There could be more elegant options than nesting a page inside a page.
5. Your last comment makes me wonder if this is another "we need this yesterday" type requests. Is that the case?

**Comment by AnaHotaling** [comment]
LOL! With all of us drowning in marathon stuff, it's not a "we need this
yesterday" thing. It's a "we discussed it several times already in the past
and Eva would like it done but not yet a top priority" thing. :)

Okay... forget embedding the page from Crowdrise. That was the easiest
answer since all of the donation widgets are there. The page is at
https://www.crowdrise.com/epicraces

If you go there, you'll see all the widgets lined up in chronological order
at the bottom of the page. Well, not all, since there are a few events I
haven't gotten to yet. The first one is the marathon widget that is
currently at the bottom of the Epic home page. We want to replace that one
sole widget on the Epic homepage with an array of all of the donation
widgets, as seen on the Crowdrise page.

We had believed, after meeting with the Crowdrise group, that THEY were
going to provide us with the code that would display all of our widgets. We
in turn would give that code to you, and you'd plug it into that two-column
space. Well, they gave us the code for the marathon widget and, when I
asked for the code that would show our multiple widgets, I was told that
this was something our developers -- you! -- would have to finagle.

Quick note that there are quite a few things Eva and I thought Crowdrise
was going to be doing for us that I ended up doing. Like creating the pages
for our events over there. But that's another story!!!

Anyhow, the widgets are already plugged into the Epic site -- they are tied
into that Fundraising button you implemented for us. Nothing new is being
introduced... it's just in a sense gathering them from all the event pages
and arranging them neatly in the space on the home page (each one will
still be on its event page, too). Eva really liked how the widgets looked
on the Crowdrise site, all neatly displayed. The Crowdrise reps showed us
the page for the NY Road Runners Club and a couple of other race companies,
and that was exactly what Eva wanted: an eye-catching graphic display of
fundraising through our events.

She doesn't want text. She wants images, since images draw attention. And
she likes how each event widget features either the event logo or a photo
from the event and how it displays the total donations to date plus a
clickable Donate button. She has wanted a Giving Back to the Community
section on the homepage for a while, and once told me that people should
know that Epic gives more to charity than Champions for Charity does. She
likes the subtlety of the widgets, since they aren't hitting people over
the head, demanding donations.

So that's what I'd like to try to do: create a display of our event
donation widgets on the home page, where that solitary widget is now. I'd
asked for that space to be wide like that because we really thought
Crowdrise was going to give us the code to allow us to recreate how the
widgets are displayed on the Epic Crowdrise page. Otherwise, that marathon
widget would've easily fit a one-column space.

What can be done? I haven't checked if I could lift the source of the Epic
Crowdrise page with Web Developer... that would make everything easy!! I
also think that, to fit all the widgets in, it'll take a lot more space on
our site unless we scale the widgets down. Putting them in full sized would
easily overwhelm the home page and make it all Frankensteiny, like you
said. Or just really unwieldy. You already explained about carousels not
being all to friendly for older displays, so carouseling the widgets is
out. A shifting grid, kind of like the table option for the VRB? Except
vertical instead of horizontal, and it would only show three widgets at a
time; when an event is done, I'd go in and drag it down to the bottom in
sort order (btw I really hadn't noticed those numbers in the VRB table
display! Glad Eric told me about them).

I'm totally open to ideas!

On Thursday, March 31, 2016, Kyle notifications@github.com wrote:

> I believe you're opening this up here now for discussion purposes,
> correct? If you're asking for something specific to be done it isn't clear
> yet. It sounds like you're trying to make a decision about how best to
> display some new information. Can I ask some questions?
> 1. Could you simply blog about Epic's giving and include prominent
>    links to donate opportunities? If not enough people read the blog you could
>    send it to your newsletter list and we could discuss ways to make the blog
>    more prominent. You could also share the post on social media and it would
>    show in the Epic News widget on the home page.
> 2. Instead of embedding something, could we just design some nice
>    looking button or new widget that describes and links to this information?
>    Embedding from an external source can be problematic for a number of
>    reasons but one big reason is that it will not look like your site. It will
>    usually look a bit out of place and unprofessional. That's because you're
>    simply pulling information from a different website that will be styled
>    like _that_ site and not _yours_. I'd hate to see the Epic Races home
>    page get cluttered and lose its cohesive style.
> 3. If we are going to add something new to the home page, do we really
>    need to _replace_ something? There aren't a limited number of spaces.
>    We can keep going down the page almost infinitely. So something new could
>    be placed above or below what's there now.
> 4. Oh, now I'm getting what you said about replacing the _widget_ with
>    the _page_. Ummm...like I said in 2 above, embedding an external page
>    will look bad and there's really not much you can do about it. It is a big
>    step towards a Frankenstein page where some elements very obviously don't
>    belong and are not cohesive with the rest. I'd recommend using caution
>    here. If there is information you'd like to present to visitors beyond what
>    is there, let's talk about it. There could be more elegant options than
>    nesting a page inside a page.
> 5. Your last comment makes me wonder if this is another "we need this
>    yesterday" type requests. Is that the case?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/realbig/RBM_TH_EPIC/issues/61#issuecomment-203895167

## 

Ana Hotaling
Director of Inbound Marketing, Epic Races
p: 734-585-7101 a: 600 S. Wagner Rd., Ann Arbor, MI 48103
s: http://www.epicraces.com e: ana@epicraces.com
http://facebook.com/epicracesevents http://epicraces http://epicraces

https://newoldstamp.com/editor/

**Comment by brashrebel** [comment]
Hmmm...well that sounds fun. So what you need us to do is...figure out a way to pull info about several crowdrise events at once and display it nicely on the home page instead of the single widget that is there now?

**Comment by AnaHotaling** [comment]
That's pretty much it!

I still haven't checked the source code. But at least I'm not brushing my
teeth and emailing you at the same time now!

On Thursday, March 31, 2016, Kyle notifications@github.com wrote:

> Hmmm...well that sounds fun. So what you need us to do is...figure out a
> way to pull info about several crowdrise events at once and display it
> nicely on the home page instead of the single widget that is there now?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/realbig/RBM_TH_EPIC/issues/61#issuecomment-203977400

## 

Ana Hotaling
Director of Inbound Marketing, Epic Races
p: 734-585-7101 a: 600 S. Wagner Rd., Ann Arbor, MI 48103
s: http://www.epicraces.com e: ana@epicraces.com
http://facebook.com/epicracesevents http://epicraces http://epicraces

https://newoldstamp.com/editor/

**Comment by AnaHotaling** [comment]
So... how are things going on this end? :)

**Comment by d4mation** [comment]
I've been looking into getting all the Crowdrise Fundraising Events onto the Homepage and outside of manually copying their code in for each one (Which would be very tedious), I've found a much better option:

[Apparently Crowdrise has an API](https://www.crowdrise.com/fundraising-events#api).

It seems like none of that is public facing (No documentation that I can find, but I also don't have an account with them to look more closely) so I can't determine whether you have access to it or not. However, if we can get access to their API, we can do a lot of cool things with the Crowdrise Widgets.

Remember how with the big widget on the Home Page we were able to change how it looked a little bit, but we were ultimately restricted by how they looked out-of-the-box?

Using the API, we could grab all the Data for each Fundraising Event and then make them look however we want, which would let us easily put them into a grid and make them match look and feel of the site much better.

**Comment by AnaHotaling** [comment]
Just sent an email asking for it!

Ana Hotaling
Director of Inbound Marketing, Epic Races
p: 734-585-7101 a: 600 S. Wagner Rd., Ann Arbor, MI 48103
s: http://www.epicraces.com e: ana@epicraces.com
http://facebook.com/epicracesevents http://epicraces http://epicraces

https://newoldstamp.com/editor/

**Comment by AnaHotaling** [comment]
Okay! Here is what I've got:

API Key: 163cc4577b22f9609ad043b7583dc09d7e88a611

API Secret: 9464dfaa103991993fe96a137330d07421d27775

API Documentation:  	CrowdRise_Event_Api_Integration_Guide_v1.3.2.pdf (link is https://cdncustom.crowdrise.com/_crowdrise/CrowdRise_Event_Api_Integration_Guide_v1.3.2.pdf)

API Sandbox:  	https://apisandbox.crowdrise.com/

HTTP username:  	 	crowdrise

HTTP password:  	apis4ndb0x

HTTP login/password: not required

**Comment by d4mation** [comment]
I deleted that comment since it had some sensitive information (API Secret), but I've got it stored safely!

**Comment by AnaHotaling** [comment]
Thanks!!!

Ana Hotaling
Director of Inbound Marketing, Epic Races
p: 734-585-7101 a: 600 S. Wagner Rd., Ann Arbor, MI 48103
s: http://www.epicraces.com e: ana@epicraces.com
http://facebook.com/epicracesevents http://epicraces http://epicraces

https://newoldstamp.com/editor/

**Comment by d4mation** [comment]
I've been building an interface for managing this and making things look nice and I realized that I'm pulling an Event for "Swim to the Moon" that has no photo or fundraising goal.

Is that one not meant to be public yet? I figure it isn't, I just want to know so I can figure out what to check for to prevent it from showing if needed.

**Comment by AnaHotaling** [comment]
Nuts! That means that Crowdrise hasn't connected with the folks from North Star Reach yet. I'll let our Crowdrise person, Annette, know. Not even the logo is in there?

**Comment by d4mation** [comment]
Yeah, the logo doesn't show for that one.

Here's a quick screenshot of things as they are right now with this custom widget. It is just using a placeholder provided by Crowdrise for that Event.

![image](https://cloud.githubusercontent.com/assets/7770631/14719997/3e948bda-07cd-11e6-91b7-12656d7cb047.png)

**Comment by AnaHotaling** [comment]
I just set a fundraising goal, then contacted Annette at Crowdrise, since
she was given all the logos and graphics for the event. She is going to
have the graphics team pull something together so that the widget isn't
completely blank. Unfortunately, she didn't give me a timeline. Would it be
possible to not pull STTM for now, and then get it when Crowdrise has
everything set up?

Ana Hotaling
Director of Inbound Marketing, Epic Races
p: 734-585-7101 a: 600 S. Wagner Rd., Ann Arbor, MI 48103
s: http://www.epicraces.com e: ana@epicraces.com
http://facebook.com/epicracesevents http://epicraces http://epicraces

https://newoldstamp.com/editor/

**Comment by d4mation** [comment]
There's not many good ways to do this based on what I have access to with this API, unfortunately.

Here's a couple Ideas:
1. I can make a box where Event IDs can be entered to be excluded. Unfortunately, I can't find any way to find these easily (Outside of the response I get from the API). The only other place I've seen the Event IDs would be in the old Widget Codes that we got from them. It gets added to the HTML IDs on some of the elements. This would work, but it would require someone to manually remove it from the exclusion list once it is ready. (The Event ID is `9993` in this example) ![image](https://cloud.githubusercontent.com/assets/7770631/14721193/2805df26-07d3-11e6-8a63-5f41dc39a4b7.png)
2. Right now in order to display the other logos correctly, I'm having to check to see if the URL for the logo contains `crowdrise.com` or not&mdash;as they normally don't and I have to add it. The placeholder image for STTM already has `crowdrise.com` which I would assume is true of any Event that is using a placeholder image. I could check to see if an Event is using a Placeholder image, and if it is, skip it. This should then automatically show STTM when it is ready.
3. Similarly, I could exclude any Event that has no Fundraising Goal. This wouldn't work for STTM now that one is set, but it could work for future Events if you'd like to filter those out.
4. We could also filter by Event Name, which would be easier than Event ID for sure. It just leads to a lot of the same problems as using Event IDs and would actually be more difficult to build a good interface for doing that.

Personally, I think a combination of 2 and 3 would be the best way to handle this.

**Comment by AnaHotaling** [comment]
I cede to whatever you think will work best!

Ana Hotaling
Director of Inbound Marketing, Epic Races
p: 734-585-7101 a: 600 S. Wagner Rd., Ann Arbor, MI 48103
s: http://www.epicraces.com e: ana@epicraces.com
http://facebook.com/epicracesevents http://epicraces http://epicraces

https://newoldstamp.com/editor/

**Comment by d4mation** [comment]
I decided to have it check for both Placeholder Images and whether the Fundraising Goal wasn't set. We can always change how that works later.

All that I've really got left to do for this is some mobile tweaks for the "1 Event Per Column" setting. What do you think so far? Is there anything that you'd say is missing or you would like to see differently? I didn't want to make it too different from what you already have so it didn't throw off visitors to your site. I have the old widget above in the screenshots for comparison purposes.

Unfortunately, there's no support in the API for something like the little pop-up window the old Widget code provided by Crowdrise does when you hit the "Donate Now" button. It instead links them to the Event's page on Crowdrise ([Like this one](https://www.crowdrise.com/AnnArborMarathon2016)) where they can then make a Donation.

Here's what the Settings look like:

![image](https://cloud.githubusercontent.com/assets/7770631/14724222/bfb1ec0c-07e2-11e6-936e-02c4c312c8f7.png)

You are able to set the number of Columns to any number between 1 and 6 and it will divide up the Events accordingly in a grid. However, at 4-6 things do get a little squished due to the length of some Event titles so I wouldn't recommend it.

You can set it to only grab the `X` most recent Events, or you can leave it at the default of `-1` which will grab all of them available on Crowdrise.

Here's a couple screenshots of it in action:

Three Events per Row. On Mobile each column is full-width.
![image](https://cloud.githubusercontent.com/assets/7770631/14723960/9119db9e-07e1-11e6-8aff-af3f9ba22e5c.png)

With each Event taking its own Row they alternate left and right
![image](https://cloud.githubusercontent.com/assets/7770631/14723952/86356996-07e1-11e6-99b9-170fd62f2677.png)

**Comment by AnaHotaling** [comment]
Perfect! And you say I can toggle on and off which events' widgets show up?

Ana Hotaling
Director of Inbound Marketing, Epic Races
p: 734-585-7101 a: 600 S. Wagner Rd., Ann Arbor, MI 48103
s: http://www.epicraces.com e: ana@epicraces.com
http://facebook.com/epicracesevents http://epicraces http://epicraces

https://newoldstamp.com/editor/

**Comment by d4mation** [comment]
Not specific events, no. Not without putting in something like I mentioned earlier where you could specify the Event ID. I was saying that you could say "Only show the 3 most recent Events" and it would show only the 3 most recent.

If you're wanting to show only 1 or 2 in particular, it may make the most sense in that use-case to use the Code provided by Crowdrise that you were using before, as I imagine you're thinking of placing a certain one on certain pages, right? This widget is meant for the Home Page and would be a bit more global than that.

**Comment by AnaHotaling** [comment]
Yeah, Eva wants all the widgets (or as many as possible) on the home page,
so that works fine! We still haven't even gotten to Turkey Trot yet... what
happens when we add new events? Do I let you know that this whole thing
needs to be updated?

Ana Hotaling
Director of Inbound Marketing, Epic Races
p: 734-585-7101 a: 600 S. Wagner Rd., Ann Arbor, MI 48103
s: http://www.epicraces.com e: ana@epicraces.com
http://facebook.com/epicracesevents http://epicraces http://epicraces

https://newoldstamp.com/editor/

**Comment by d4mation** [comment]
It should happen automatically! 

The Server holds a cache of the Home Page for performance reasons so it could take a day or so to show up. We can manually flush the cache as-needed though.

**Comment by AnaHotaling** [comment]
Woohoo! Excellent! Thanks, Eric!!!

Ana Hotaling
Director of Inbound Marketing, Epic Races
p: 734-585-7101 a: 600 S. Wagner Rd., Ann Arbor, MI 48103
s: http://www.epicraces.com e: ana@epicraces.com
http://facebook.com/epicracesevents http://epicraces http://epicraces

https://newoldstamp.com/editor/

**Comment by d4mation** [comment]
This should be all set to go on your Live Site! I've configured it to use 3 columns for now, but as I said before it can be easily changed for any configuration.
